it/uk/gov/hmrc/repositories/CacheRepositorySpec.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.repositories
+
+import org.scalatest.{FreeSpec, MustMatchers}
+import play.api.libs.json.Json
+import play.api.test.Helpers.running
+import uk.gov.hmrc.estates.repositories.CacheRepository
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class CacheRepositorySpec extends FreeSpec with MustMatchers with TransformIntegrationTest {
+
+  "a playback repository" - {
+    "must be able to store and retrieve a payload" in {
+
+      val application = applicationBuilder.build()
+
+      running(application) {
+        getConnection(application).map { connection =>
+
+          dropTheDatabase(connection)
+
+          val repository = application.injector.instanceOf[CacheRepository]
+
+          val storedOk = repository.set("UTRUTRUTR", "InternalId", data)
+          storedOk.futureValue mustBe true
+
+          val retrieved = repository.get("UTRUTRUTR", "InternalId")
+            .map(_.getOrElse(fail("The record was not found in the database")))
+
+          retrieved.futureValue mustBe data
+
+          dropTheDatabase(connection)
+        }.get
+      }
+    }
+  }
```

Comment:
There was a safer implementation that I came up with in the maintain-other-individuals-frontend microservice. Can you bring this over as it means we don't have to do `.get`.

Reply:
maintain-other-individuals-frontend has no tests for the repository that I can see?